### DISCUSSION TITLE
Allow `return void`

### DISCUSSION THREAD

**Discussion by YT-GameMaker** [dotnet/csharplang#7862] [Language Ideas] [answered]
```csharp
void Func1() { }

void Func2()
{
    if (...)
        return Func1();

    // This is equivalent to
    if (...)
    {
        Func1();
        return;
    }

    ...
}
```

**Comment by HaloFour** [comment, marked as answer]
See: #135

**Reply by YT-GameMaker** [reply]
Ha, it has always been there. I think their discussion has deviated from the main point. Treat it as a simplified syntax; there's no need to try to understand it from a type perspective. Syntax is just syntax, and if it is useful for us, then it can exist.

If we have to explain it from a type perspective, `void` means nothing. So why do methods still need `void` as a return type? Since `void` represents nothing, then `return void` is equivalent to `return`, and there's nothing difficult to understand about it.

**Reply by huoyaoyuan** [reply]
This is just how C-family and many other languages designed.
If `void` is allowed in generic parameter, `return T` when T is `void` will be more realistic.

**Reply by huoyaoyuan** [reply]
> Treat it as a simplified syntax

How does it simplify? We are not very interested in removing a pair of braces.

**Reply by YT-GameMaker** [reply]
Yes, just reducing the number of lines and braces, but this is very important. Imagine how many programmers there are in the world, and they might be writing code like this every day. They could potentially interrupt their programming at any time, wondering, "Why can't I do it this way?" And this will continue indefinitely.

**Reply by huoyaoyuan** [reply]
I believe it's much less popular comparing to what expression body (=>) solves.

**Comment by Captain-Quack** [comment]
Perhaps this feature could work sort of like the `unit` type in F#?

**Comment by ghost** [comment]
I hope ```?:``` operator can support return ```void```.
```
        public static void M(bool condition)
        {
            return condition == false ? Console.WriteLine("False") : Console.WriteLine("True");
        }
```
Of course, I know that ```void``` is not a type. Maybe ```M``` should be
```
        public static void M(bool condition)
        {
            condition == false ? Console.WriteLine("False") : Console.WriteLine("True");
        }
```

**Comment by elanatframework** [comment]
Hello

It would be great if we could return a void method (and even a value) in another void method.

**Example:**

Please see the following codes in C# language:
```csharp
    public void PageLoad(HttpContext context)
    {
        if (!string.IsNullOrEmpty(context.Request.Form["btn_Delete"]))
        {
            btn_Delete_Click(context);
            return;
        }

        if (!string.IsNullOrEmpty(context.Request.Form["btn_GoToEdit"]))
        {
            btn_GoToEdit_Click(context);
            return;
        }

        // Do something
    }

    private void btn_Delete_Click(HttpContext context)
    {
        // Do something
    }

    private void btn_GoToEdit_Click(HttpContext context)
    {
        // Do something
    }
```

Now compare the codes below with the codes above:
```csharp
    public void PageLoad(HttpContext context)
    {
        if (!string.IsNullOrEmpty(context.Request.Form["btn_Delete"]))
            return btn_Delete_Click(context);
            
        if (!string.IsNullOrEmpty(context.Request.Form["btn_GoToEdit"]))
            return btn_GoToEdit_Click(context);

        // Do something
    }

    private void btn_Delete_Click(HttpContext context)
    {
        // Do something
    }

    private void btn_GoToEdit_Click(HttpContext context)
    {
        // Do something
    }
```

If this feature is added to C#, the codes will be cleaner.

**Reply by iam3yal** [reply]
@elanatframework Just my opinion, generally speaking I'm in favor of this feature but I think that the examples here are not compelling enough or that the motivation here is lacking because there are many aspects that concerns `void` beyond reducing lines of code and in many cases one can use `if else if else` and avoid having to `return` each time.

**Reply by elanatframework** [reply]
You are right, the benefits are more than code simplification. I wrote that even if there is some amount, it can be returned.

**Examples of controller class methods in MVC:**

```csharp
public IActionResult Get()
{
    if (MyCondiction)
        return View("~/User/Index.cshtml");

    // Do something
}
```

According to the code above, the controller class methods in MVC return various return values ​​to IActionResult. In some cases, string is used instead of IActionResult.

It would be better if the View method was not returned. But according to the codes below, the code lines increase.

```csharp
public void Get()
{
    if (MyCondiction)
    {
        View("/User/Index.cshtml");
        return;
    }

    // Do something
}
```

IActionResult return type is hard to understand for newbies. Maybe if it was possible to return values ​​to void functions, the controller class methods like below would be simpler.

```csharp
public void Get()
{
    if (MyCondiction)
        return View("~/User/Index.cshtml");

    // Do something
}
```
